### PR TITLE
[next batch] Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')

### PR DESCRIPTION
**Pull request by zero-24**
Pull Request for Issue #10399 .
#### Summary of Changes

Always use htmlspecialchars($str, ENT_COMPAT, 'UTF-8')
#### Testing Instructions

Please review or test every file. Thanks

### CODE REVIEW

File: components/com_users/views/profile/tmpl/default_core.php [hunk 1]
```diff
@@ -20,13 +20,13 @@
 			<?php echo JText::_('COM_USERS_PROFILE_NAME_LABEL'); ?>
 		</dt>
 		<dd>
-			<?php echo $this->data->name; ?>
+			<?php echo htmlspecialchars($this->data->name, ENT_COMPAT, 'UTF-8'); ?>
```
Comment (brianteeman) [inline components/com_users/views/profile/tmpl/default_core.php]: showing my ignorance perhaps but why did you add it to this one. surely it has already been filtered

File: layouts/joomla/edit/frontediting_modules.php [hunk 2]
```diff
@@ -43,7 +43,7 @@
 	'\\1 jmoddiv" data-jmodediturl="' . $editUrl . '" data-target="' . $target . '" data-jmodtip="'
 	.	JHtml::tooltipText(
 			JText::_('JLIB_HTML_EDIT_MODULE'),
-			htmlspecialchars($mod->title) . '<br />' . sprintf(JText::_('JLIB_HTML_EDIT_MODULE_IN_POSITION'), htmlspecialchars($position)),
+			htmlspecialchars($mod->title, ENT_COMPAT, 'UTF-8') . '<br />' . sprintf(JText::_('JLIB_HTML_EDIT_MODULE_IN_POSITION'), htmlspecialchars($position)),
```
Comment (andrepereiradasilva) [inline layouts/joomla/edit/frontediting_modules.php]: you missed one here.

File: modules/mod_menu/tmpl/default_url.php [hunk 3]
```diff
@@ -49,4 +49,4 @@
 	$attributes['onclick'] = "window.open(this.href, 'targetWindow', '" . $options . "'); return false;";
 }
 
-echo JHtml::_('link', JFilterOutput::ampReplace(htmlspecialchars($item->flink)), $linktype, $attributes);
+echo JHtml::_('link', JFilterOutput::ampReplace(htmlspecialchars($item->flink), ENT_COMPAT, 'UTF-8'), $linktype, $attributes);
```
Comment (andrepereiradasilva) [inline modules/mod_menu/tmpl/default_url.php]: Parentheses is in the wrong place.

**Comment by wojsmol** [comment]
@zero-24 see https://github.com/zero-24/joomla-cms/pull/15

**Comment by andrepereiradasilva** [comment]
seems fine to me now.

File: components/com_users/views/profile/tmpl/default_core.php (hunk 1 above)
Comment (zero-24) [inline components/com_users/views/profile/tmpl/default_core.php]: reverted.

File: administrator/templates/isis/html/modules.php [hunk 4]
```diff
@@ -50,11 +50,11 @@ function modChrome_well($module, &$params, &$attribs)
 		$moduleTag     = $params->get('module_tag', 'div');
 		$bootstrapSize = (int) $params->get('bootstrap_size');
 		$moduleClass   = ($bootstrapSize) ? ' span' . $bootstrapSize : '';
-		$headerTag     = htmlspecialchars($params->get('header_tag', 'h2'));
+		$headerTag     = htmlspecialchars($params->get('header_tag', 'h2'), ENT_COMPAT, 'UTF-8');
```
Comment (brianteeman) [inline administrator/templates/isis/html/modules.php]: is this in the right place?

Comment (wojsmol) [inline administrator/templates/isis/html/modules.php]: yes

Comment (brianteeman) [inline administrator/templates/isis/html/modules.php]: ok thanks

**Comment by andrepereiradasilva** [comment]
I have tested this item :white_check_mark: successfully on c64219525ee4e0a8b30332a4052ea1edd2281cd8<br /><br />on code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10424">issues.joomla.org/joomla-cms/10424</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on c64219525ee4e0a8b30332a4052ea1edd2281cd8<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10424">issues.joomla.org/joomla-cms/10424</a>.</sub>

**Comment by brianteeman** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/10424">issues.joomla.org/joomla-cms/10424</a>.</sub>

**Comment by roland-d** [comment]
Thanks everybody
